ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/LedgerTimeAwareCommandExecutor.scala

```diff
@@ -47,7 +47,19 @@ private[apiserver] final class LedgerTimeAwareCommandExecutor(
   )(implicit
       executionContext: ExecutionContext,
       loggingContext: LoggingContext,
-  ): Future[Either[ErrorCause, CommandExecutionResult]] =
+  ): Future[Either[ErrorCause, CommandExecutionResult]] = {
+    def retryOrError(newCommands: Commands) = {
+      if (retriesLeft > 0) {
```

Comment:
So there is no reason to assume that the two restart mechanisms will never interleave on the same request.

Reply:
That doesn't seem to be a problem does it?

Reply:
What do you mean?

Reply:
I mean a "contract spent in the meantime" and having to "advance the time" can alternate on the same request.
I was just thinking aloud.